provider/hybrid.go

```diff
@@ -0,0 +1,152 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/logging"
+	rpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+// Hybrid provider struct
+type dockerHybridProvider struct {
```

Comment:
Hmm. So I'm a little confused. In the schema code this PR uses ExtraTypes functionality that is built into the pulumi-terraform-bridge codebase. Does that codebase provide no facilities for intermixing providers? Or perhaps we could put this functionality in pulumi/pulumi? This seems generally quite useful!

Reply:
ExtraTypes/extra resources essentially only provides codegen support at the moment. I do think we can push some of this into the bridge but this seems like a reasonable starting point for now.

Reply:
The ExtraTypes/ExtraResources lets us write them to the schema, but that is the current limit here. Sounds like it might be nice to add hybrid muxing into pulumi/pulumi in the future though!